pkg/controllers/customeradmin/group_mapping_test.go

```diff
@@ -15,15 +15,30 @@ import (
 )
 
 func TestFromMSGraphGroup(t *testing.T) {
+	// see main function for explanations
 	userv1 := fakeuserv1.NewSimpleClientset(
+		// case 1
 		&v1.User{
 			ObjectMeta: metav1.ObjectMeta{
-				Name: "live.com#foo@bar.com",
+				Name: "owner@home.com",
```

Comment:
can you add a test for "bob_owner@home.com", I have done this locally and it fails. 
The code below changes the first _ instead of the second one "bob@owner_home.com" - this then fails the email validation.
```golang
email := strings.Replace(s, "_", "@", 1)
```

Reply:
should work no. I hope :)